ライブラリの署名を変更する

karronoli opened the pull request:
修正したライブラリをOPENLOGI Organization傘下で管理するため、ライブラリの署名を機械的に変更する。
修正案は Napp/salesforce-api-wrapper#1 や Napp/salesforce-api-wrapper#2 などを取り込む予定です。

karronoli committed:
名前空間の変更.

$ git ls-files | xargs sed -i '' 's/Napp/OPENLOGI/g'
$ git ls-files | xargs sed -i '' 's/napp/openlogi/g'

karronoli committed:
著者を変更

haradakunihiko commented:
あまり自信なく書いてますが、
ライセンスは元の作成者のものを残さないといけないんじゃないんでしたっけ。
なんかいいソースがないですが

https://stackoverflow.com/questions/20427285/how-do-i-adapt-an-existing-licence-from-a-forked-project-giving-all-due-credit-t/20427471#20427471
https://mozxxx.hatenadiary.org/entry/20110529/p4

別の例)
https://github.com/tcocca/acts_as_follower/blob/master/MIT-LICENSE

ただライセンスを含めていれば改変OKだったと思うので、面倒であればそのまま使ってもいいはず？？

karronoli commented:
たしかに。Napp 版に辿れるようにうまいこと残しておきたいです。
GitHub 上では Napp 版からのフォークした形になりましたが、本当の大本のライブラリはこっちでした。 https://packagist.org/packages/crunch-accounting/salesforce-api
表記的に両方を残すにはどうすればいいんだろう…

karronoli commented:
replace の指定は使ったほうが良さそう。
https://getcomposer.org/doc/04-schema.md#replace

あと README からもリンクするようにしてみます。

karronoli committed:
composer の replace を指定.

https://getcomposer.org/doc/04-schema.md#replace
```
replace

Lists packages that are replaced by this package. This allows you to fork a package, publish it under a different name with its own version numbers, while packages requiring the original package continue to work with your fork because it replaces the original package.
```

karronoli committed:
元々のライブラリへのリンクを追加

karronoli commented:
Napp 版のライセンスファイルをなかったことにするのは気が引けてきました。リネームして残しておきます。

karronoli committed:
napp/salesforce-api のライセンスファイルを別名で残す

haradakunihiko commented:
あざまーす。良いかと思います！
